hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/SqlFilterProjectTest.java

```diff
@@ -88,8 +88,8 @@ public void test_valuesSelectExpressionFilterExpression() {
                         + ") AS t (a, b) "
                         + "WHERE a + b + 0 + CAST('1' AS TINYINT) > 4",
                 asList(
-                        new Row((byte) -2),
-                        new Row((byte) -4)
+                        new Row((short) -2),
```

Comment:
Why the type changed here?

Reply:
Because now we infer the types in the same way as IMDG. [See here](https://github.com/hazelcast/hazelcast/blob/d0ee94c2fb888ceb37a6ea7928044326da300a99/hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/misc/HazelcastArithmeticOperatorUtils.java#L43-L52).

Literal `1` is `TINYINT(1)`. `1+1` is `TINYINT(1) + TINYINT(1) -> TINYINT(2)`. However, with a column reference in `a - b`, it's `TINYINT(7) - TINYINT(7) -> SMALLINT(8)`. In 4.1 it probably was TINYINT. Therefore the result is `short`.